<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Introduction to the msigdbr package • msigdbr</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/readable/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to the msigdbr package">
<meta property="og:description" content="msigdbr">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">msigdbr</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">7.1.1.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/msigdbr-intro.html">Introduction to the msigdbr package</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/igordot/msigdbr/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="msigdbr-intro_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Introduction to the msigdbr package</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/igordot/msigdbr/blob/master/vignettes/msigdbr-intro.Rmd"><code>vignettes/msigdbr-intro.Rmd</code></a></small>
      <div class="hidden name"><code>msigdbr-intro.Rmd</code></div>

    </div>

    
    
<div id="overview" class="section level2">
<h2 class="hasAnchor">
<a href="#overview" class="anchor"></a>Overview</h2>
<p>Performing pathway analysis is a common task in genomics and there are many available software tools, many of which are R-based. Depending on the tool, it may be necessary to import the pathways into R, translate genes to the appropriate species, convert between symbols and IDs, and format the object in the required way.</p>
<p>The <code>msigdbr</code> R package provides Molecular Signatures Database (MSigDB) gene sets typically used with the Gene Set Enrichment Analysis (GSEA) software:</p>
<ul>
<li>in an R-friendly tidy format (a data frame in a “long” format with one gene per row)</li>
<li>for multiple frequently studied model organisms (human, mouse, rat, pig, zebrafish, fly, yeast, etc.)</li>
<li>as both gene symbols and NCBI/Entrez Gene IDs (for better compatibility with pathway enrichment tools)</li>
<li>that can be used in a script without requiring additional external files</li>
</ul>
<p>Please be aware that the homologs were computationally predicted for distinct genes. The full pathways may not be well conserved across species.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>The package can be installed from <a href="https://cran.r-project.org/package=msigdbr">CRAN</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span><span class="op">(</span><span class="st">"msigdbr"</span><span class="op">)</span></pre></div>
</div>
<div id="usage" class="section level2">
<h2 class="hasAnchor">
<a href="#usage" class="anchor"></a>Usage</h2>
<p>Load package.</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/igordot/msigdbr">msigdbr</a></span><span class="op">)</span></pre></div>
<p>Retrieve the gene sets data frame. In this example, for the hallmark collection.</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="va">h_gene_sets</span> <span class="op">=</span> <span class="fu"><a href="../reference/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Homo sapiens"</span>, category <span class="op">=</span> <span class="st">"H"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">h_gene_sets</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 16</span>
<span class="co">#&gt;   gs_name gs_id gs_cat gs_subcat gs_pmid gs_geoid gs_exact… gs_url gs_descr… human_ge…</span>
<span class="co">#&gt;   &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;    </span>
<span class="co">#&gt; 1 HALLMA… M5905 H      ""        ""      ""       ""        ""     Genes up… ABCA1    </span>
<span class="co">#&gt; 2 HALLMA… M5905 H      ""        ""      ""       ""        ""     Genes up… ABCB8    </span>
<span class="co">#&gt; 3 HALLMA… M5905 H      ""        ""      ""       ""        ""     Genes up… ACAA2    </span>
<span class="co">#&gt; 4 HALLMA… M5905 H      ""        ""      ""       ""        ""     Genes up… ACADL    </span>
<span class="co">#&gt; 5 HALLMA… M5905 H      ""        ""      ""       ""        ""     Genes up… ACADM    </span>
<span class="co">#&gt; 6 HALLMA… M5905 H      ""        ""      ""       ""        ""     Genes up… ACADS    </span>
<span class="co">#&gt; # … with 6 more variables: species_name &lt;chr&gt;, species_common_name &lt;chr&gt;,</span>
<span class="co">#&gt; #   entrez_gene &lt;int&gt;, gene_symbol &lt;chr&gt;, sources &lt;chr&gt;, num_sources &lt;dbl&gt;</span></pre></div>
<p>There is a helper function to show the available species.</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="fu"><a href="../reference/msigdbr_show_species.html">msigdbr_show_species</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt;  [1] "Bos taurus"               "Caenorhabditis elegans"   "Canis lupus familiaris"  </span>
<span class="co">#&gt;  [4] "Danio rerio"              "Drosophila melanogaster"  "Gallus gallus"           </span>
<span class="co">#&gt;  [7] "Homo sapiens"             "Mus musculus"             "Rattus norvegicus"       </span>
<span class="co">#&gt; [10] "Saccharomyces cerevisiae" "Sus scrofa"</span></pre></div>
<p>All gene sets in the database can be retrieved without specifying a collection/category.</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="va">all_gene_sets</span> <span class="op">=</span> <span class="fu"><a href="../reference/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Homo sapiens"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">all_gene_sets</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 16</span>
<span class="co">#&gt;   gs_name gs_id gs_cat gs_subcat gs_pmid gs_geoid gs_exact… gs_url gs_descr… human_ge…</span>
<span class="co">#&gt;   &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;    </span>
<span class="co">#&gt; 1 AAACCA… M126… C3     MIR:MIR_… ""      ""       ""        ""     Genes ha… ABCC4    </span>
<span class="co">#&gt; 2 AAACCA… M126… C3     MIR:MIR_… ""      ""       ""        ""     Genes ha… ABRAXAS2 </span>
<span class="co">#&gt; 3 AAACCA… M126… C3     MIR:MIR_… ""      ""       ""        ""     Genes ha… ACTN4    </span>
<span class="co">#&gt; 4 AAACCA… M126… C3     MIR:MIR_… ""      ""       ""        ""     Genes ha… ACVR1    </span>
<span class="co">#&gt; 5 AAACCA… M126… C3     MIR:MIR_… ""      ""       ""        ""     Genes ha… ADAM9    </span>
<span class="co">#&gt; 6 AAACCA… M126… C3     MIR:MIR_… ""      ""       ""        ""     Genes ha… ADAMTS5  </span>
<span class="co">#&gt; # … with 6 more variables: species_name &lt;chr&gt;, species_common_name &lt;chr&gt;,</span>
<span class="co">#&gt; #   entrez_gene &lt;int&gt;, gene_symbol &lt;chr&gt;, sources &lt;chr&gt;, num_sources &lt;dbl&gt;</span></pre></div>
<p>The <code><a href="../reference/msigdbr.html">msigdbr()</a></code> function output is a data frame and can be manipulated using more standard methods.</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="va">all_gene_sets</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">gs_cat</span> <span class="op">==</span> <span class="st">"H"</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 16</span>
<span class="co">#&gt;   gs_name gs_id gs_cat gs_subcat gs_pmid gs_geoid gs_exact… gs_url gs_descr… human_ge…</span>
<span class="co">#&gt;   &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;    </span>
<span class="co">#&gt; 1 HALLMA… M5905 H      ""        ""      ""       ""        ""     Genes up… ABCA1    </span>
<span class="co">#&gt; 2 HALLMA… M5905 H      ""        ""      ""       ""        ""     Genes up… ABCB8    </span>
<span class="co">#&gt; 3 HALLMA… M5905 H      ""        ""      ""       ""        ""     Genes up… ACAA2    </span>
<span class="co">#&gt; 4 HALLMA… M5905 H      ""        ""      ""       ""        ""     Genes up… ACADL    </span>
<span class="co">#&gt; 5 HALLMA… M5905 H      ""        ""      ""       ""        ""     Genes up… ACADM    </span>
<span class="co">#&gt; 6 HALLMA… M5905 H      ""        ""      ""       ""        ""     Genes up… ACADS    </span>
<span class="co">#&gt; # … with 6 more variables: species_name &lt;chr&gt;, species_common_name &lt;chr&gt;,</span>
<span class="co">#&gt; #   entrez_gene &lt;int&gt;, gene_symbol &lt;chr&gt;, sources &lt;chr&gt;, num_sources &lt;dbl&gt;</span></pre></div>
<p>Check the available collections and sub-collections.</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="va">all_gene_sets</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html">distinct</a></span><span class="op">(</span><span class="va">gs_cat</span>, <span class="va">gs_subcat</span><span class="op">)</span> <span class="op">%&gt;%</span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html">arrange</a></span><span class="op">(</span><span class="va">gs_cat</span>, <span class="va">gs_subcat</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 19 x 2</span>
<span class="co">#&gt;    gs_cat gs_subcat       </span>
<span class="co">#&gt;    &lt;chr&gt;  &lt;chr&gt;           </span>
<span class="co">#&gt;  1 C1     ""              </span>
<span class="co">#&gt;  2 C2     "CGP"           </span>
<span class="co">#&gt;  3 C2     "CP"            </span>
<span class="co">#&gt;  4 C2     "CP:BIOCARTA"   </span>
<span class="co">#&gt;  5 C2     "CP:KEGG"       </span>
<span class="co">#&gt;  6 C2     "CP:PID"        </span>
<span class="co">#&gt;  7 C2     "CP:REACTOME"   </span>
<span class="co">#&gt;  8 C3     "MIR:MIR_Legacy"</span>
<span class="co">#&gt;  9 C3     "MIR:MIRDB"     </span>
<span class="co">#&gt; 10 C3     "TFT:GTRD"      </span>
<span class="co">#&gt; 11 C3     "TFT:TFT_Legacy"</span>
<span class="co">#&gt; 12 C4     "CGN"           </span>
<span class="co">#&gt; 13 C4     "CM"            </span>
<span class="co">#&gt; 14 C5     "BP"            </span>
<span class="co">#&gt; 15 C5     "CC"            </span>
<span class="co">#&gt; 16 C5     "MF"            </span>
<span class="co">#&gt; 17 C6     ""              </span>
<span class="co">#&gt; 18 C7     ""              </span>
<span class="co">#&gt; 19 H      ""</span></pre></div>
<p>Retrieve mouse C2 (curated) CGP (chemical and genetic perturbations) gene sets.</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="va">cgp_gene_sets</span> <span class="op">=</span> <span class="fu"><a href="../reference/msigdbr.html">msigdbr</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Mus musculus"</span>, category <span class="op">=</span> <span class="st">"C2"</span>, subcategory <span class="op">=</span> <span class="st">"CGP"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">cgp_gene_sets</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 6 x 16</span>
<span class="co">#&gt;   gs_name gs_id gs_cat gs_subcat gs_pmid gs_geoid gs_exact… gs_url gs_descr… human_ge…</span>
<span class="co">#&gt;   &lt;chr&gt;   &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;   &lt;chr&gt;    &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;     &lt;chr&gt;    </span>
<span class="co">#&gt; 1 ABBUD_… M1423 C2     CGP       145761… ""       Table 2   ""     Genes do… AHNAK    </span>
<span class="co">#&gt; 2 ABBUD_… M1423 C2     CGP       145761… ""       Table 2   ""     Genes do… ANKRD40  </span>
<span class="co">#&gt; 3 ABBUD_… M1423 C2     CGP       145761… ""       Table 2   ""     Genes do… ARID1A   </span>
<span class="co">#&gt; 4 ABBUD_… M1423 C2     CGP       145761… ""       Table 2   ""     Genes do… BCKDHB   </span>
<span class="co">#&gt; 5 ABBUD_… M1423 C2     CGP       145761… ""       Table 2   ""     Genes do… C16orf89 </span>
<span class="co">#&gt; 6 ABBUD_… M1423 C2     CGP       145761… ""       Table 2   ""     Genes do… CAPN9    </span>
<span class="co">#&gt; # … with 6 more variables: species_name &lt;chr&gt;, species_common_name &lt;chr&gt;,</span>
<span class="co">#&gt; #   entrez_gene &lt;int&gt;, gene_symbol &lt;chr&gt;, sources &lt;chr&gt;, num_sources &lt;dbl&gt;</span></pre></div>
</div>
<div id="pathway-enrichment-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#pathway-enrichment-analysis" class="anchor"></a>Pathway enrichment analysis</h2>
<p>The <code>msigdbr</code> gene sets data frame can be used with many pathway analysis packages.</p>
<p>Use the gene sets data frame for <code>clusterProfiler</code> (for genes as Entrez Gene IDs).</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="va">msigdbr_t2g</span> <span class="op">=</span> <span class="va">msigdbr_df</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">gs_name</span>, <span class="va">entrez_gene</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu">enricher</span><span class="op">(</span>gene <span class="op">=</span> <span class="va">gene_ids_vector</span>, TERM2GENE <span class="op">=</span> <span class="va">msigdbr_t2g</span>, <span class="va">...</span><span class="op">)</span></pre></div>
<p>Use the gene sets data frame for <code>clusterProfiler</code> (for genes as gene symbols).</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="va">msigdbr_t2g</span> <span class="op">=</span> <span class="va">msigdbr_df</span> <span class="op">%&gt;%</span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html">select</a></span><span class="op">(</span><span class="va">gs_name</span>, <span class="va">gene_symbol</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span>
<span class="fu">enricher</span><span class="op">(</span>gene <span class="op">=</span> <span class="va">gene_symbols_vector</span>, TERM2GENE <span class="op">=</span> <span class="va">msigdbr_t2g</span>, <span class="va">...</span><span class="op">)</span></pre></div>
<p>Use the gene sets data frame for <code>fgsea</code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="va">msigdbr_list</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html">split</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">msigdbr_df</span><span class="op">$</span><span class="va">gene_symbol</span>, f <span class="op">=</span> <span class="va">msigdbr_df</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span>
<span class="fu">fgsea</span><span class="op">(</span>pathways <span class="op">=</span> <span class="va">msigdbr_list</span>, <span class="va">...</span><span class="op">)</span></pre></div>
<p>Use the gene sets data frame for <code>GSVA</code>.</p>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="va">msigdbr_list</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html">split</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">msigdbr_df</span><span class="op">$</span><span class="va">gene_symbol</span>, f <span class="op">=</span> <span class="va">msigdbr_df</span><span class="op">$</span><span class="va">gs_name</span><span class="op">)</span>
<span class="fu">gsva</span><span class="op">(</span>gset.idx.list <span class="op">=</span> <span class="va">msigdbr_list</span>, <span class="va">...</span><span class="op">)</span></pre></div>
</div>
<div id="questions-and-concerns" class="section level2">
<h2 class="hasAnchor">
<a href="#questions-and-concerns" class="anchor"></a>Questions and concerns</h2>
<p><strong>Which version of MSigDB was used?</strong></p>
<p>This package was generated with MSigDB v7.1 (released March 2020). The MSigDB version is used as the base of the package version. You can check the installed version with <code><a href="https://rdrr.io/r/utils/packageDescription.html">packageVersion("msigdbr")</a></code>.</p>
<p><strong>Can I download the gene sets directly from MSigDB instead of using this package?</strong></p>
<p>Yes. You can then import the GMT files (with <code>getGmt()</code> from the <code>GSEABase</code> package, for example). The GMTs only include the human genes, even for gene sets generated from mouse data. If you are not working with human data, you then have to convert the MSigDB genes to your organism or your genes to human.</p>
<p><strong>Can I convert between human and mouse genes just by adjusting gene capitalization?</strong></p>
<p>That will work for most genes, but not all.</p>
<p><strong>Can I convert human genes to any organism myself instead of using this package?</strong></p>
<p>Yes. A popular method is using the <code>biomaRt</code> package. You may still end up with dozens of homologs for some genes, so additional cleanup may be helpful.</p>
<p><strong>Aren’t there already other similar tools?</strong></p>
<p>There are a few other resources that and provide some of the functionality and served as an inspiration for this package. <a href="http://ge-lab.org/#/data">Ge Lab Gene Set Files</a> has GMT files for many species. <a href="http://bioinf.wehi.edu.au/software/MSigDB/">WEHI</a> provides MSigDB gene sets in R format for human and mouse, but the genes are provided only as Entrez IDs and each collection is a separate file. <a href="https://github.com/stephenturner/msigdf">MSigDF</a> is based on the WEHI resource, so it provides the same data, but converted to a more tidyverse-friendly data frame. When <code>msigdbr</code> was initially released, all of them were multiple releases behind the latest version of MSigDB, so they are possibly no longer maintained.</p>
</div>
<div id="details" class="section level2">
<h2 class="hasAnchor">
<a href="#details" class="anchor"></a>Details</h2>
<p>The Molecular Signatures Database (MSigDB) is a collection of gene sets originally created for use with the Gene Set Enrichment Analysis (GSEA) software.</p>
<p>Gene homologs are provided by HUGO Gene Nomenclature Committee at the European Bioinformatics Institute which integrates the orthology assertions predicted for human genes by eggNOG, Ensembl Compara, HGNC, HomoloGene, Inparanoid, NCBI Gene Orthology, OMA, OrthoDB, OrthoMCL, Panther, PhylomeDB, TreeFam and ZFIN. For each human equivalent within each species, only the ortholog supported by the largest number of databases is used.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Igor Dolgalev.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
